dev/archery/archery/crossbow/core.py

```diff
@@ -361,6 +361,24 @@ def signature(self):
         return pygit2.Signature(self.user_name, self.user_email,
                                 int(time.time()))
 
+    @property
+    def default_branch_name(self):
+        default_branch_name = os.getenv("DEFAULT_BRANCH")
```

Comment:
Hmm... doesn't this environment variable risk clashing with other utilities using the same variable name?
Ideally we don't need an environment variable, or we can make it more specific (for example `ARCHERY_DEFAULT_BRANCH`).

Reply:
Since the git repositories do not always have a remote head reference, we use an environment variable as a secondary way to get the default branch name. After the changes from this code review, the order will be:
1. Get the default branch from the repository's remote head reference (most likely to be correct, if present in the repository)
2. Check for an environment variable that will contain the default branch name (a way for the user to either override the first heuristic, if needed)
3. Default to `master` and warn that the other two heuristics did not result in a default branch name
The addition of support for the environment variable also reduces the reliance on the hard-coded default branch name value in step 3. 

I will change the environment variable to a more specific name, as you suggested, `ARCHERY_DEFAULT_BRANCH`

Reply:
Ok, that sounds like a good strategy to me.